File: benchmark/c/cpw/find_minimum_sum_factors_number/find_minimum_sum_factors_number.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int find_minimum_sum_factors_number ( int num ) ;
int find_minimum_sum_factors_number ( int num ) {
  int sum = 0;
  for ( int i = 2;
  i * i <= num;
  i ++ ) {
    while ( num % i == 0 ) {
      sum += i;
      num /= i;
    }
  }
  sum += num;
  return sum;
}

File: translation_result/else_bms/fail/c/cpw/find_minimum_sum_factors_number/gpt4/BaseRepair/find_minimum_sum_factors_number.rs

fn find_minimum_sum_factors_number(mut num: i32) -> i32 {
    if num < 0 {
        return num;
    }

    let mut sum: i32 = 0;
    let mut i = 2;

    while i * i <= num {
        while num % i == 0 {
            sum = sum.wrapping_add(i);
            num /= i;
        }
        i += 1;
    }

    if num > 1 {
        sum = sum.wrapping_add(num);
    }

    sum
}
